contracts/integrations/test/internal-integration-tests/exchange_wrapper_test.ts

```diff
@@ -0,0 +1,1003 @@
+                    weth: deployment.tokens.weth,
+                },
+            },
+            {},
+        );
+
+        await blockchainBalances.updateBalancesAsync();
+
+        initialLocalBalances = LocalBalanceStore.create(blockchainBalances);
+
+        wethAssetData = assetDataUtils.encodeERC20AssetData(deployment.tokens.weth.address);
+    });
+
+    beforeEach(async () => {
+        localBalances = LocalBalanceStore.create(initialLocalBalances);
+    });
+
+    function addFillResults(a: FillResults, b: FillResults): void {
+        a.makerAssetFilledAmount = a.makerAssetFilledAmount.plus(b.makerAssetFilledAmount);
+        a.takerAssetFilledAmount = a.takerAssetFilledAmount.plus(b.takerAssetFilledAmount);
+        a.makerFeePaid = a.makerFeePaid.plus(b.makerFeePaid);
+        a.takerFeePaid = a.takerFeePaid.plus(b.takerFeePaid);
+        a.protocolFeePaid = a.protocolFeePaid.plus(b.protocolFeePaid);
+    }
+
+    function simulateFill(signedOrder: SignedOrder, expectedFillResults: FillResults, shouldUseWeth: boolean): void {
+        // taker -> maker
+        localBalances.transferAsset(
+            taker,
+            maker,
+            expectedFillResults.takerAssetFilledAmount,
+            signedOrder.takerAssetData,
+        );
+
+        // maker -> taker
+        localBalances.transferAsset(
+            maker,
+            taker,
+            expectedFillResults.makerAssetFilledAmount,
+            signedOrder.makerAssetData,
+        );
+
+        // maker -> feeRecipient
+        localBalances.transferAsset(
+            maker,
+            feeRecipient,
+            expectedFillResults.makerFeePaid,
+            signedOrder.makerFeeAssetData,
+        );
+
+        // taker -> feeRecipient
+        localBalances.transferAsset(
+            taker,
+            feeRecipient,
+            expectedFillResults.takerFeePaid,
+            signedOrder.takerFeeAssetData,
+        );
+
+        // taker -> protocol fees
+        if (shouldUseWeth) {
+            localBalances.transferAsset(
+                taker,
+                deployment.staking.stakingProxy.address,
+                expectedFillResults.protocolFeePaid,
+                wethAssetData,
+            );
+        } else {
+            localBalances.sendEth(taker, deployment.staking.stakingProxy.address, expectedFillResults.protocolFeePaid);
+        }
+    }
+
+    function maximumBatchFillResults(signedOrders: SignedOrder[]): FillResults {
+        return signedOrders
+            .map(signedOrder => ({
+                makerAssetFilledAmount: signedOrder.makerAssetAmount,
+                takerAssetFilledAmount: signedOrder.takerAssetAmount,
+                makerFeePaid: signedOrder.makerFee,
+                takerFeePaid: signedOrder.takerFee,
+                protocolFeePaid: DeploymentManager.protocolFee,
+            }))
+            .reduce(
+                (totalFillResults, currentFillResults) => ({
```

Comment:
Deleting this function (as per the above comment). It might be cool to add a `sumFillResults` reference function, but I think that is out-of-scope for this PR.